pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -308,3 +334,62 @@ func createValidImageVersionLabel(image string) string {
 	}
 	return cephImageVersion
 }
+
+func waitForPvcToExpandWithTimeout(ctx context.Context, client client.Client, pvcResizeMap map[string]pvcResize, namespace string, pvcResizeWaitTimeOut time.Duration) {
+	currentTime := time.Now()
+	if pvcResizeWaitTimeOut == 0 {
+		pvcResizeWaitTimeOut = defaultWaitTimeoutForHealthyOSD
+	}
+	timeout := currentTime.Add(pvcResizeWaitTimeOut)
+	allPvcResize := false
+	for !allPvcResize {
+		if time.Now().After(timeout) {
+			logger.Errorf("timed out waiting for PVC size to be updated, osds may need to be restarted manually after resize")
+			return
+		}
+		allPvcResize = checkAllPvcResize(ctx, client, namespace, pvcResizeMap)
+		if !allPvcResize {
+			// wait for 5 seconds before checking again
+			time.Sleep(5 * time.Second)
+		}
+	}
+}
+
+func checkAllPvcResize(ctx context.Context, client client.Client, namespace string, pvcResizeMap map[string]pvcResize) bool {
```

Comment:
There could be hundreds or even thousands of OSDs, so if we are checking all of them every 5 seconds until they are all done it could be inefficient. How about constructing a map for the completed resizes? As they are resized, they would move from one map to the other. This will also cut down on logging the same status for a PVC many times if this method has a lot of retries. We always have to think about scale with OSDs.

Reply:
Added a simpler bool resize variable to track it